src/open_inwoner/configurations/bootstrap/kic.py

```diff
@@ -0,0 +1,190 @@
+from django.conf import settings
+
+import requests
+from django_setup_configuration.configuration import BaseConfigurationStep
+from django_setup_configuration.exceptions import SelfTestFailed
+from zgw_consumers.constants import APITypes, AuthTypes
+from zgw_consumers.models import Service
+
+from open_inwoner.openklant.clients import build_client
+from open_inwoner.openklant.models import OpenKlantConfig
+from open_inwoner.utils.api import ClientError
+
+
+class KlantenAPIConfigurationStep(BaseConfigurationStep):
+    """
+    Configure the required Service to establish a connection with the Klanten API
+    """
+
+    verbose_name = "Klanten API configuration"
+    required_settings = [
+        "KIC_CONFIG_KLANTEN_API_ROOT",
+        "KIC_CONFIG_KLANTEN_API_CLIENT_ID",
+        "KIC_CONFIG_KLANTEN_API_SECRET",
+    ]
+    enable_setting = "KIC_CONFIG_ENABLE"
+
+    def is_configured(self) -> bool:
+        return bool(
+            Service.objects.filter(api_root=settings.KIC_CONFIG_KLANTEN_API_ROOT)
+        )
+
+    def configure(self):
+        organization = settings.OIP_ORGANIZATION or settings.ENVIRONMENT
+        org_label = f"Open Inwoner {organization}".strip()
+
+        # 1. Create Klanten API service
+        klanten_service, created = Service.objects.update_or_create(
+            api_root=settings.KIC_CONFIG_KLANTEN_API_ROOT,
+            defaults={
+                "label": "Klanten API",
+                "api_type": APITypes.kc,
+                "oas": settings.KIC_CONFIG_KLANTEN_API_ROOT,
+                "auth_type": AuthTypes.zgw,
+                "client_id": settings.KIC_CONFIG_KLANTEN_API_CLIENT_ID,
+                "secret": settings.KIC_CONFIG_KLANTEN_API_SECRET,
+                "user_id": settings.KIC_CONFIG_KLANTEN_API_CLIENT_ID,
+                "user_representation": org_label,
+            },
+        )
+        if not created:
+            klanten_service.oas = settings.KIC_CONFIG_KLANTEN_API_ROOT
+            klanten_service.client_id = settings.KIC_CONFIG_KLANTEN_API_CLIENT_ID
+            klanten_service.secret = settings.KIC_CONFIG_KLANTEN_API_SECRET
+            klanten_service.user_id = settings.KIC_CONFIG_KLANTEN_API_CLIENT_ID
+            klanten_service.user_representation = org_label
+            klanten_service.save()
```

Comment:
Ah you're right, I thought that `defaults` were only used if the instance is created, but now I see that there is `create_defaults` for that https://docs.djangoproject.com/en/5.0/ref/models/querysets/#update-or-create. I'll remove these if blocks.

Reply:
Oh whoops, in that case we do actually need the if block, because without it existing instances won't be updated in case we run the command with `--overwrite`

Reply:
Are you sure? The kwargs select the instance (eg: api_root): if it exists it gets updated with `defaults`, if it doesn't it gets created with the values from both kwargs and `defaults`.

I don't see anything here that would do something different when `--overwrite` or whatever would be be specified. It is a plain `update_or_create()` (not a `get_or_create()` with an condition on `created` and `--overwrite`)

Reply:
Ah I was confusing it with `get_or_create` :facepalm: